<?xml version="1.0"?>
<launch>
    <!-- Simulation arguments -->
    <arg name="sim" default="true" doc="If true, launch the gazebo simulation" />
    <arg name="sim_headless" default="false" doc="If true, launch the gazebo simulation" />
    <arg name="gazebo_world" default="$(find gazebo_ros)/worlds/empty_world.world"/>

    <!-- RViz arguments -->
    <arg name="launch_rviz" default="true"/>
    <arg name="rviz_config_file" default="$(find hrii_task_board_fsm)/rviz/main_fsm.rviz"/>
    
    <!-- Left robot arguments -->
    <arg name="left_robot_id" default="franka_left"/>
    <arg name="left_robot_controller" default="cartesian_impedance_example_controller"/>
    <arg name="left_robot_init_pose_x" default="0"/>
    <arg name="left_robot_init_pose_y" default="-0.45"/>
    <arg name="left_robot_init_pose_z" default="0"/>
    <arg name="left_robot_init_pose_yaw" default="1.5707"/>

    <!-- Right robot arguments -->
    <arg name="right_robot_id" default="franka_right"/>
    <arg name="right_robot_controller" default="cartesian_impedance_example_controller"/>
    <arg name="right_robot_init_pose_x" default="0"/>
    <arg name="right_robot_init_pose_y" default="0.45"/>
    <arg name="right_robot_init_pose_z" default="0"/>
    <arg name="right_robot_init_pose_yaw" default="-1.5707"/>

    <!-- Launch left robot -->
    <group ns="$(arg left_robot_id)">
        <include file="$(find hrii_task_board_fsm)/launch/launch_robot.xml" pass_all_args="true">
            <arg name="arm_id" value="$(arg left_robot_id)"/>
            <arg name="x" value="$(arg left_robot_init_pose_x)"/>
            <arg name="y" value="$(arg left_robot_init_pose_y)"/>
            <arg name="z" value="$(arg left_robot_init_pose_z)"/>
            <arg name="yaw" value="$(arg left_robot_init_pose_yaw)"/>
            <arg name="controller" value="$(arg left_robot_controller)"/>
        </include>
    </group>

    <!-- Launch right robot -->
    <!-- <group ns="$(arg right_robot_id)">
        <include file="$(find hrii_task_board_fsm)/launch/launch_robot.xml" pass_all_args="true">
            <arg name="arm_id" value="$(arg right_robot_id)"/>
            <arg name="x" value="$(arg right_robot_init_pose_x)"/>
            <arg name="y" value="$(arg right_robot_init_pose_y)"/>
            <arg name="z" value="$(arg right_robot_init_pose_z)"/>
            <arg name="yaw" value="$(arg right_robot_init_pose_yaw)"/>
            <arg name="controller" value="$(arg right_robot_controller)"/>
        </include>
    </group> -->
    
    <include if="$(arg sim)" file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="$(arg gazebo_world)"/>
        <arg name="paused" value="true"/>
        <arg name="gui" value="$(eval not arg('sim_headless'))"/>
        <arg name="use_sim_time" value="true"/>
    </include>
    
    <group ns="fsm">
        <node pkg="hrii_task_board_fsm" type="main_fsm" name="main_fsm" output="screen" respawn="false">
            <param name="left_robot_id" value="$(arg left_robot_id)"/>
            <param name="right_robot_id" value="$(arg right_robot_id)"/>
        </node>

        <node pkg="hrii_task_board_fsm" type="press_button_fsm" name="press_button_fsm" output="screen" respawn="false"/>
        
    </group>
    <node if="$(arg launch_rviz)" pkg="rviz" type="rviz" output="screen" name="rviz" args="-f $(arg left_robot_id)_link0 -d $(arg rviz_config_file) --splash-screen $(find hrii_task_board_fsm)/splash.png"/>
     
</launch>
